web/lib/src/helpers/lists.dart

```diff
@@ -69,3 +70,44 @@ class JSImmutableListWrapper<T extends JSObject, U extends JSObject>
   @override
   U elementAt(int index) => this[index];
 }
+
+/// A wrapper for live node lists. `NodeList` and `HTMLCollection` that are
+/// [live](https://developer.mozilla.org/en-US/docs/Web/API/NodeList#live_vs._static_nodelists)
+/// can be safely modified at runtime. This requires an instance of `P`, a
+/// container that elements would be added to or removed from.
+class JSLiveNodeListWrapper<P extends Node, T extends JSObject, U extends Node>
+    extends JSImmutableListWrapper<T, U> {
+  final P parentNode;
+
+  JSLiveNodeListWrapper(this.parentNode, super.original);
+
+  @override
+  set length(int value) {
+    if (value > length) {
+      throw UnsupportedError('Cannot add null to live node List.');
+    }
+    for (var i = length - 1; i >= value; i--) {
+      parentNode.removeChild(_jsList.item(i));
+    }
+  }
+
+  @override
+  void operator []=(int index, U value) {
+    RangeError.checkValidRange(index, null, length);
+    parentNode.replaceChild(value, _jsList.item(index));
+  }
+
+  @override
+  void add(U element) {
+    // `ListMixin` implementation only works for lists that allow `null`.
+    parentNode.appendChild(element);
+  }
+
+  @override
+  void removeRange(int start, int end) {
+    RangeError.checkValidRange(start, end, length);
+    for (var i = 0; i < end - start + 1; i++) {
+      parentNode.removeChild(this[start]);
```

Comment:
Should this be `start + i`?

Reply:
Ah yes, the list is getting readjusted as you're removing elements. And yeah, I don't think the `end--` makes this any less confusing so this is fine.